src/commands/synthetics/README.md

```diff
@@ -157,86 +106,53 @@ For example:
-| failOnMissingTests   | `DATADOG_SYNTHETICS_FAIL_ON_MISSING_TESTS`   |
-| failOnTimeout        | `DATADOG_SYNTHETICS_FAIL_ON_TIMEOUT`         |
-| files                | `DATADOG_SYNTHETICS_FILES`                   |
-| jUnitReport          | `DATADOG_SYNTHETICS_JUNIT_REPORT`            |
-| publicIds            | `DATADOG_SYNTHETICS_PUBLIC_IDS`              |
-| selectiveRerun       | `DATADOG_SYNTHETICS_SELECTIVE_RERUN`         |
-| subdomain            | `DATADOG_SUBDOMAIN`                          |
-| testSearchQuery      | `DATADOG_SYNTHETICS_TEST_SEARCH_QUERY`       |
-| tunnel               | `DATADOG_SYNTHETICS_TUNNEL`                  |
-| latest               | `DATADOG_SYNTHETICS_LATEST`                  |
-| mobileApplicationId  | `DATADOG_SYNTHETICS_MOBILE_APPLICATION_ID`   |
+### Environment Variables
+
+In addition to the global configuration file, you can configure all properties using environment variables. If a property is defined in both the global configuration file and as an environment variable, the environment variable takes precedence.
+
+Example:
 
+```bash
+export DATADOG_SITE=datadoghq.com
+```
 
 ### Command line options
 
-If the organization uses a custom sub-domain to access Datadog, this needs to be set in the `DATADOG_SUBDOMAIN` environment variable or in the global configuration file under the `subdomain` key in order to properly display the test results URL. 
+The CLI provides another way to set options and configure the behavior of datadog-ci. These options will override the global configuration file and environment variables.
 
-For example, if the URL used to access Datadog is `myorg.datadoghq.com`, set the environment variable to `myorg`:
+Example:
 
 ```bash
-export DATADOG_SUBDOMAIN="myorg"
+yarn datadog-ci synthetics run-tests --public-id pub-lic-id1
 ```
 
-You can use `DATADOG_SYNTHETICS_LOCATIONS` to override the locations where your tests run. Locations should be separated with `;`. The configuration in [test files](#test-files) takes precedence over other overrides.
+The priority of the 3 forms of configuration is as follows:
 
-```bash
-export DATADOG_SYNTHETICS_LOCATIONS="aws:us-east-1;aws:us-east-2"
+```yml
+Global Config < Environment variables < CLI parameters
 ```
 
-### API
+### Using datadog-ci as a library
 
-By default, `datadog-ci` runs at the root of the working directory and looks for `{,!(node_modules)/**/}*.synthetics.json` files (every file ending with `.synthetics.json`, except for those in the `node_modules` folder) to find a [test configuration file](#test-files).
-The default file name for the [global configuration file](#global-configuration-file-options) is `datadog-ci.json`. If you use this name for your global configuration file, you may omit the `--config` flag.
+You can also use the `datadog-ci` package as a library in your Node.js application. To do so, import the package and use the `execute()` function for either `runTests` or `uploadApplication`.
```

Comment:
I tried to detail a bit this section, but couldn't get everything right. I think we should remove this section and cut another card to flesh this out with a proper example and instructions.

``````suggestion
You can also use the `datadog-ci` package as a library in your Node.js application to trigger tests. To do so, import the package from the Synthetics `run-tests` command and call the `executeWithDetails()` function.

``` javascript
import { executeWithDetails } from '@datadog/datadog-ci/dist/commands/synthetics/run-tests-lib';

const { results, summary } = await runTests.executeWithDetails(...)
```
``````

Reply:
Why are you suggesting to import from `@datadog/datadog-ci/dist/commands/synthetics/run-tests-lib`?

We exported a [`synthetics` object](https://github.com/DataDog/datadog-ci/blob/fdf21c69bc79f35fe62b4356e881c3bdcfb29fd3/src/index.ts#L2) from the main index file for that purpose. The `executeWithDetails()` function was made for us internally because [we didn't want to break the current lib API](https://github.com/DataDog/datadog-ci/pull/1128#issuecomment-1838136065).

So I'd rather document the current "official" way to use the lib, or nothing (and refine the API as part of a card).

The "official" way is basically:
- Either the user only wants to call one "do-it-all" function, and they can do:
  ```ts
  import {synthetics} from '@datadog/datadog-ci'

  await synthetics.execute(
    {
      apiKey: '',
      appKey: '',
      publicIds: ['aaa-bbb-ccc'],
    },
    {}
  )
  ```
- Or they can use `synthetics.executeTests()` along with the exported utils to build their own logic (example taken [from our github action](https://github.com/DataDog/synthetics-ci-github-action/blob/fe4e6106a85c3439e2d90977cf4bda7f58183cd8/src/main.ts#L5-L41)):
  ```ts
  import {synthetics, utils} from '@datadog/datadog-ci'

  const config: synthetics.RunTestsCommandConfig = {
    // their config
  }

  const reporter = synthetics.utils.getReporter([new synthetics.DefaultReporter({context: process})])
  const startTime = Date.now()

  try {
    const {results, summary} = await synthetics.executeTests(reporter, config)
    const orgSettings = await synthetics.utils.getOrgSettings(reporter, config)

    synthetics.utils.renderResults({
      config,
      orgSettings,
      reporter,
      results,
      startTime,
      summary,
    })

    synthetics.utils.reportExitLogs(reporter, config, {results})

    const exitReason = synthetics.utils.getExitReason(config, {results})
    if (exitReason !== 'passed') {
      console.error(`Datadog Synthetics tests failed: ${printSummary(summary, config)}`)
    } else {
      console.info(`Datadog Synthetics tests succeeded: ${printSummary(summary, config)}`)
    }
  } catch (error) {
    synthetics.utils.reportExitLogs(reporter, config, {error})

    const exitReason = synthetics.utils.getExitReason(config, {error})
    if (exitReason !== 'passed') {
      console.error('Running Datadog Synthetics tests failed.')
    }
  }
  ```